File: AnalyticAlways.AngularTest.Domain/Article.cs
namespace AnalyticAlways.AngularTest.Domain
{
    public class Article
    {
        protected Article()
        { }

        public Article(string description, double price)
        {
            Description = description;
            Price = price;
        }

        public int Id { get; private set; }

        public string Description { get; private set; }

        public double Price { get; private set; }
    }
}

File: angular-test2/src/app/articles/shared/article.ts
export class Article {

    public id: number;
    public description: string;
    public price: number;

    public static parse(json: any): Article {
        return new Article(
            json.id,
            json.description,
            json.price);
    }

    public static parseCollection(json: any): Article[] {
        return json.map(item => {
            return Article.parse(item);
        });
    }

    constructor(
        id: number,
        description: string,
        price: number
    ) {
        this.id = id;
        this.description = description;
        this.price = price;
    }
}
